src/Model.php

```diff
@@ -1465,32 +1422,26 @@ public function tryLoadAny()
      *
      * @return $this
      */
-    public function loadBy(string $field_name, $value)
+    public function loadBy(string $fieldName, $value)
     {
-        // store
-        $field = $this->getField($field_name);
-        $system = $field->system;
-        $default = $field->default;
+        $field = $this->getField($fieldName);
 
-        // add condition and load record
-        $this->addCondition($field_name, $value);
+        $scopeBak = $this->scope;
+        $systemBak = $field->system;
+        $defaultBak = $field->default;
 
         try {
-            $this->loadAny();
-        } catch (\Exception $e) {
-            // restore
-            array_pop($this->conditions);
-            $field->system = $system;
-            $field->default = $default;
+            // add condition to cloned scope and try to load record
+            $this->scope = clone $this->scope;
+            $this->addCondition($field, $value);
 
-            throw $e;
+            $this->loadAny();
```

Comment:
slighly better CS would be to return here and remove `return $this;` below

2x